import "common.proto";
import "player_response.proto";

message BattleUnit{
	required int32 no = 1;  //战斗单位ID
	optional int32 quality = 2;  //战斗单位品质
	optional int32 normal_skill = 3;  //战斗单位普通技能
	optional int32 rage_skill = 4;  //战斗单位怒气技能
	optional float hp = 5; // 战斗单位血量
	optional float atk = 6; //战斗单位攻击
	optional float physical_def = 7;//战斗单位物理防御
	optional float magic_def = 8;  //战斗单位魔法防御
	optional float hit = 9; //战斗单位命中率
	optional float dodge = 10;  //战斗单位闪避率
	optional float cri = 11;  //战斗单位暴击率
	optional float cri_coeff = 12; //战斗单位暴击伤害系数
	optional float cri_ded_coeff = 13;  //战斗单位暴伤减免系数
	optional float block = 14;  //战斗单位格挡率
	optional bool is_boss = 15; //是否是boss
}

message BattleUnitGrop{
	repeated BattleUnit group = 1;
}

message Stage{
	required int32 stage_id = 1;  //关卡编号
	optional int32 attacks = 2;  //攻击次数
	optional int32 state = 3;  //关卡状态 -1:开启没打过 0：输  1：赢
}

message StageAward{
	required int32 chapter_id = 1; //章节编号
	repeated int32 award = 2;  //奖励数组 -1:奖励没达成 0：奖励达成没有领取 1：已经领取
	optional int32 dragon_gift = 3; //龙纹奖励 -1:奖励没达成 0：奖励达成没有领取 1：已经领取
}


//请求关卡信息 command:901
message StageInfoRequest{
	required int32 stage_id = 1; //关卡编号 0: 全部信息
}

message StageInfoResponse{
	repeated Stage stage = 1;

}

//请求章节奖励信息 command:902
message ChapterInfoRequest{
	required int32 chapter_id = 1; //章节编号 0：全部信息
}

message ChapterInfoResponse{
	repeated StageAward stage_award = 1;
}

// 进入关卡战斗 command:903
message StageStartRequest{
	required int32 stage_id = 1; //关卡编号
}

message StageStartResponse{
	required CommonResponse res = 1;
	optional int32 drop_num = 2;  //掉落数量
	repeated BattleUnit red = 4; //红方数据 自己
	repeated BattleUnitGrop blue = 5; //对方数据
}


// 关卡结算 command:904
message StageSettlementRequest{
	required int32 stage_id = 1; //关卡编号
	required int32 result = 2;  //输赢
}
message StageSettlementResponse{
	optional GameResourcesResponse drops = 2;
}
